src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -600,11 +600,12 @@ public void convict(InetAddressAndPort endpoint, double phi)
      * This method is used to mark a node as shutdown; that is it gracefully exited on its own and told us about it
      * @param endpoint endpoint that has shut itself down
      */
+    @Deprecated // can remove once 4.x is not supported
     protected void markAsShutdown(InetAddressAndPort endpoint)
     {
         checkProperThreadForStateMutation();
         EndpointState epState = endpointStateMap.get(endpoint);
-        if (epState == null)
+        if (epState == null || epState.isStateEmpty())
```

Comment:
Is this enough to prevent the issue without sending the endpoint state?  If so maybe we can do this in lower branches?

Reply:
this blocks the known corruption issue, so back porting this to 4.x should be safe and fix this specific issue, but won't fix the other issue we talked about, that host replacement is missing TOKENS.  Given the jira is only about the NPE that feels like a good solution to fix the most branches without any protocol level change